dev/io.openliberty.http.monitor_fat/fat/src/io/openliberty/http/monitor/fat/JSPApplicationTest.java

```diff
@@ -0,0 +1,131 @@
+                        .add(new FileAsset(new File("test-applications/JspApp/resource/configured.jsp")), "/configured.jsp")
+                        .add(new FileAsset(new File("test-applications/JspApp/resource/unconfigured.jsp")), "/unconfigured.jsp")
+                        .add(new FileAsset(new File("test-applications/JspApp/resource/default.html")), "/default.html")
+                        .add(new FileAsset(new File("test-applications/JspApp/resource/Testhtml.html")), "Testhtml.html")
+                        .addPackage(
+                                    "io.openliberty.http.monitor.fat.jspApp");
+
+        // test-applications\JspApp\src\io\openliberty\http\monitor\fat\jspApp\resource
+        ShrinkHelper.exportDropinAppToServer(server, testWAR,
+                                             DeployOptions.SERVER_ONLY);
+
+        server.startServer();
+
+        //Read to run a smarter planet
+        server.waitForStringInLogUsingMark("CWWKF0011I");
+        server.setMarkToEndOfLog();
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        //catch if a server is still running.
+        if (server != null && server.isStarted()) {
+            server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWPMI2006W", "CWMMC0013E", "CWWKG0033W", "SRVE0315E");
+        }
+    }
+
+    @Test
+    public void jsp_noWebXmlJSP() throws Exception {
+
+        assertTrue(server.isStarted());
+
+        String route = CONTEXT_ROOT + "/unconfigured.jsp";
+        String expectedRoute = CONTEXT_ROOT + "/\\*.jsp";
+        String requestMethod = HttpMethod.GET;
+        String responseStatus = "200";
+
+        String res = requestHttpServlet(route, server, requestMethod);
+
+        assertTrue(validatePrometheusHTTPMetric(getVendorMetrics(server), expectedRoute, responseStatus, requestMethod));
+
+    }
+
+    @Test
+    public void jsp_configredWebXmlJSP() throws Exception {
+
+        assertTrue(server.isStarted());
+
+        String route = CONTEXT_ROOT + "/configured";
+        String requestMethod = HttpMethod.GET;
+        String responseStatus = "200";
+
+        String res = requestHttpServlet(route, server, requestMethod);
+
+        assertTrue(validatePrometheusHTTPMetric(getVendorMetrics(server), route, responseStatus, requestMethod));
+
+    }
+
+    @Test
+    public void jsp_defaultHTML() throws Exception {
+
+        assertTrue(server.isStarted());
+
+        String route = CONTEXT_ROOT;
+        String expectedRoute = CONTEXT_ROOT + "/";
+        String requestMethod = HttpMethod.GET;
+        String responseStatus = "200";
+
+        String res = requestHttpServlet(route, server, requestMethod);
+
+        assertTrue(validatePrometheusHTTPMetric(getVendorMetrics(server), expectedRoute, responseStatus, requestMethod));
+
+        //Request an explicity HTML page, will also return 304, we'll now check count
```

Comment:
Should the `responseStatus` in line 124 be 304?

Reply:
Oh, should've deleted this.

From manual testing, 304 occurs because it was loading a cached version.

Reply:
Fixed in https://github.com/OpenLiberty/open-liberty/pull/28565/commits/3b17a4d12b48f93656ea09ed58209ff55e7c6843